src/webapp/reports/autogenerated-forms/GridWithCatOptionCombosViewModel.ts

```diff
@@ -0,0 +1,93 @@
+import _ from "lodash";
+import { Section, Texts } from "../../../domain/common/entities/DataForm";
+import { DataElement } from "../../../domain/common/entities/DataElement";
+
+export interface Grid {
+    id: string;
+    name: string;
+    columns: Column[];
+    rows: Row[];
+    toggle: Section["toggle"];
+    texts: Texts;
+}
+
+interface SubSectionGrid {
+    name: string;
+    dataElements: DataElement[];
+}
+
+interface Column {
+    name: string;
+    items: Array<{ row: Row; dataElement: DataElement | undefined }>;
+}
+
+interface Row {
+    groupName: string;
+    rows: {
+        deName: string;
+        name: string;
+    }[];
+}
+
+const separator = " - ";
+
+export class GridWithCatOptionCombosViewModel {
+    static get(section: Section): Grid {
+        const subsections = _(section.dataElements)
+            .flatMap(dataElement => {
+                const cocNames = dataElement.categoryOptionCombos.map(coc => coc.name);
+                return cocNames.flatMap(coc => ({
+                    ...dataElement,
+                    cocId: dataElement.categoryCombos.categoryOptionCombos.find(c => c.name === coc)?.id || "cocId",
```

Comment:
I don't understand why are we returning `cocId` if we don't found a `categoryOptionCombos`? Since we are sending this ID to the dataValues endpoint we should make sure to always send the right id. As an alternative we can save undefined and then do a `compact` so we ignore the whole item.